### PR TITLE
Tortoise ORM support

### PR DESCRIPTION
**Pull request by prostomarkeloff**
#54

### CODE REVIEW

**Comment by frankie567** [comment]
Thank you for that @prostomarkeloff! Sounds good.

Before I review in further details, could you make a pass of `make format`? :)

**Comment by prostomarkeloff** [comment]
Also, i think i should add docs :smile:

File: fastapi_users/db/tortoise.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from fastapi_users.db import BaseUserDatabase
+from fastapi_users.models import BaseUserDB
+
+
+class BaseUserModel:
+    id: str = fields.TextField(pk=True, generated=False)
+    email: str = fields.CharField(index=True, unique=True, null=False, max_length=255)
+    hashed_password: str = fields.CharField(null=False, max_length=255)
+    is_active: bool = fields.BooleanField(default=True, null=False)
+    is_superuser: bool = fields.BooleanField(default=False, null=False)
```
Comment (frankie567) [inline fastapi_users/db/tortoise.py]: Unless I'm mistaken, I don't think it makes sense to have type annotation here. Those properties are of type `TextField/BooleanField/...`, not `str/bool`.

File: tests/test_db_tortoise.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+@pytest.fixture
+async def tortoise_user_db() -> AsyncGenerator[TortoiseUserDatabase, None]:
+    DATABASE_URL = "sqlite://./test.db"
+
+    await Tortoise.init(
+        db_url=DATABASE_URL, modules={"models": ["tests.test_db_tortoise"]}
+    )
+    await Tortoise.generate_schemas()
+    yield TortoiseUserDatabase(User)
```
Comment (frankie567) [inline tests/test_db_tortoise.py]: I would put a new line before `yield` to quickly identify it

File: fastapi_users/db/tortoise.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    class Meta:
+        table = "user"
+
+
+class TortoiseUserDatabase(BaseUserDatabase):
+    def __init__(self, user_model: Type[Model]):
+        self._model = user_model
+
```
Comment (frankie567) [inline fastapi_users/db/tortoise.py]: Could you declare the property in the class and without the underscore?

```py
class TortoiseUserDatabase(BaseUserDatabase):

    model: Type[Model]

    def __init__(self, model: Type[Model]):
        self.model = model
```

File: fastapi_users/db/tortoise.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    async def get_by_email(self, email: str) -> Optional[BaseUserDB]:
+        try:
+            user = await self._model.filter(email=email).first()
+        except DoesNotExist:
+            user = None
+        return BaseUserDB.from_orm(user) if user else None
+
+    async def create(self, user: BaseUserDB) -> BaseUserDB:
+        model = self._model(**user.dict())
+        await model.save()
```
Comment (frankie567) [inline fastapi_users/db/tortoise.py]: I don't really like the variable being named `model`. Could we do it in one line with the `create` method of Tortoise?

```py
await self.model.create(**user.dict())
```

File: fastapi_users/db/tortoise.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        return user
+
+    async def update(self, user: BaseUserDB) -> BaseUserDB:
+        usr = user.create_update_dict_superuser()
+        await self._model.filter(id=user.id).update(**usr)
```
Comment (frankie567) [inline fastapi_users/db/tortoise.py]: You should always use `.dict()` in DB adapter. This method is for the superuser routes to update the user according to the logged-user rights. So you could directly do:

```py
await self._model.filter(id=user.id).update(**user.dict())
```

Comment (prostomarkeloff) [reply fastapi_users/db/tortoise.py]: If use this method, id will be included to request. Tortoise ORM will raise exception about change the PK.

File: fastapi_users/db/tortoise.py (hunk 1 above)
Comment (prostomarkeloff) [reply fastapi_users/db/tortoise.py]: Yes, my mistake.

**Comment by prostomarkeloff** [comment]
Hmmm, i really don't know why tests don't pass. On my local machine i have same behaviour.

**Comment by prostomarkeloff** [comment]
If use it:
```py
    async def create(self, user: BaseUserDB) -> BaseUserDB:
        model = self.model(**user.dict())
        await model.save()
        return user
```
instead of just await, user will be created.

**Comment by frankie567** [comment]
Hey @prostomarkeloff! I didn't forget you but I have other priorities at the moment :) I'll have a look at your PR this weekend ;)

**Comment by frankie567** [comment]
Hi @prostomarkeloff :) Could you enable [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) ? I've made some tweaks to your PR that I would like to push :)

**Comment by prostomarkeloff** [comment]
> Hi @prostomarkeloff :) Could you enable [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) ? I've made some tweaks to your PR that I would like to push :)

Hey! It already is enabled.
image

**Comment by frankie567** [comment]
@prostomarkeloff My bad! I had a hard time figuring out how I could push to your fork, GitHub documentation is not very clear about this ;)

Anyway, let's merge this 👍 When I've finished #42 (which should be very soon), I'll do a release :)

Thank you for your work!

**Comment by frankie567** [comment]
@all-contributors add @prostomarkeloff  for doc, code

**Pull request merged**